src/main/java/com/microsoft/azure/functions/worker/broker/CoreTypeResolver.java

```diff
@@ -79,19 +90,32 @@ private static String getBindingNameFromAnnotation(Annotation annotation) {
 		return null;
 	}
 
-	private static String getBindingNameFromCustomBindingAnnotation(CustomBinding customBindingAnnotation) {
+	private static String getBindingNameFromCustomBindingAnnotation(Annotation customBindingAnnotation, Method name) {
 		try {
-			return customBindingAnnotation.name();
+			return (String) name.invoke(customBindingAnnotation);
 		} catch (Exception ex) {
 			// Ignore
 			return null;
 		}
 	}
 
 	static String getBindingNameAnnotation(Parameter param) {
-		BindingName paramAnnotation = param.getDeclaredAnnotation(BindingName.class);
-		if (paramAnnotation != null) {
-			return paramAnnotation.value();
+		Annotation bindingNameAnnotation = null;
+		for (Annotation item : param.getAnnotations()){
+			if (item.toString().contains("com.microsoft.azure.functions.annotation.BindingName")){
+				bindingNameAnnotation = item;
+				break;
+			}
+		}
+		if (bindingNameAnnotation != null) {
+			String returnValue = null;
+			try {
+				Method value = bindingNameAnnotation.annotationType().getMethod("value");
+				returnValue = (String) value.invoke(bindingNameAnnotation);
+			} catch (Exception ex) {
```

Comment:
We may need to raise an exception or warning we can track just in case this case happens